src/spikeinterface/widgets/unit_summary.py

```diff
@@ -107,15 +107,18 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
         # and use custum grid spec
         fig = self.figure
         nrows = 2
-        ncols = 3
+        ncols = 2
         if sorting_analyzer.has_extension("correlograms") or sorting_analyzer.has_extension("spike_amplitudes"):
             ncols += 1
+        if sorting_analyzer.has_extension("waveforms"):
+            ncols += 1
         if sorting_analyzer.has_extension("spike_amplitudes"):
             nrows += 1
         gs = fig.add_gridspec(nrows, ncols)
+        col_counter = 0
 
         if sorting_analyzer.has_extension("unit_locations"):
-            ax1 = fig.add_subplot(gs[:2, 0])
+            ax1 = fig.add_subplot(gs[:2, col_counter])
```

Comment:
ATM this is `ax1` and the next axis is `ax2` but if `unit_locations` is not found then `ax2` will actually be the first axis. Is it worth naming these by their plot type? e.g. `unit_locations_axis` (but maybe this is confusing in other ways as you loose any sense of order)